src/utils.js

```diff
@@ -47,14 +75,24 @@ function readFiles (glob) {
  * Read the files from a glob, but then instead of resolving the
  * Promise with an Array of objects (@see readFiles), resolve with a
  * single object; each file's contents is keyed by its filename run
- * through keyname().
+ * through optional `keyFn(filePath, options)`` (default: keyname).
+ * Will pass other options on to readFiles and keyFn
  *
+ * @param {glob}
+ * @param {Object} options (all optional):
+ *  - keyFn
+ *  - contentFn
+ *  - stripNumbers
+ * @return {Promise} resolving to {Object} of keyed file contents
  */
-function readFilesKeyed (glob, preserveNumbers = false) {
-  return readFiles(glob).then(allFileData => {
+function readFilesKeyed (glob, options = {}) {
+  const {
+    keyFn = (path, options) => keyname(path, options)
+  } = options;
+  return readFiles(glob, options).then(allFileData => {
     const keyedFileData = new Object();
     for (var aFile of allFileData) {
-      keyedFileData[keyname(aFile.path, preserveNumbers)] = aFile.contents;
+      keyedFileData[keyFn(aFile.path, options)] = aFile.contents;
```

Comment:
This will default to using `keyname` to derive keys for file data, but can be overridden. Not sure if this is _useful_ but enh.